packages/pages/src/dev/server/middleware/indexPage.ts

```diff
@@ -129,6 +162,54 @@ export const indexPage =
     }
   };
 
+const createSystemEventFuncList = (
+  systemEventFuncs: Record<string, PluginSource>
+) => {
+  return Object.keys(systemEventFuncs).length
+    ? `<div class="list">
+      <div class="list-title">System Events</div>
+      <ul>${Object.values(systemEventFuncs).reduce(
+        (funcAccumulator, source) => {
+          return (
+            funcAccumulator +
+            `<li>
+            ${
+              source.event === EventType.ON_PAGE_GENERATE
```

Comment:
Could we do something a bit safer here? In the future, if a new event type is added, it could be easy to forget to update this logic.

Reply:
This logic should be removed in the next few days once Spruce finishes the item for passing the event display name in `pluginFileMap.json`. We would directly render that here instead of conditioning on event type. I can add a `TODO` comment to call attention to it